packages/manager/apps/container/src/core/product-nav-reshuffle/ProductNavReshuffleProvider.jsx

```diff
@@ -1,71 +1,98 @@
+
   const [isLoading, setIsLoading] = useState(true);
-  const [isBeta, setIsBeta] = useState(false);
-  const [askBeta, setAskBeta] = useState(false);
 
-  let pnrContext = useContext(ProductNavReshuffleContext);
+  // onboarding
+  const [onboardingOpenedState, setOnboardingOpenedState] = useState(
+    ONBOARDING_OPENED_STATE_ENUM.CLOSED,
+  );
 
-  const updateBetaChoice = async (accept = false) => {
-    return reketInstance
-      .put(`/me/preferences/manager/${preferenceKey}`, {
-        key: preferenceKey,
-        value: accept ? 'true' : 'false',
-      })
-      .then((result) => {
-        if (accept === false) {
-          // @TODO open new tab for survey
-        }
-        window.location.reload(false);
-        return result;
-      });
+  const openOnboarding = () => {
+    setOnboardingOpenedState(
+      onboardingHelper.getNextOpenedState(onboardingOpenedState),
+    );
   };
 
-  const createBetaChoice = async (accept = false) => {
-    return reketInstance
-      .post(`/me/preferences/manager`, {
-        key: preferenceKey,
-        value: accept ? 'true' : 'false',
-      })
-      .then((result) => {
-        window.location.reload(false);
-        return result;
-      });
+  const startOnboarding = () => {
+    setOnboardingOpenedState(ONBOARDING_OPENED_STATE_ENUM.WALKME);
+  };
+
+  const closeOnboarding = (onboardingStatus) => {
+    setOnboardingOpenedState(ONBOARDING_OPENED_STATE_ENUM.CLOSED);
+
+    return onboardingHelper.updatePreference({
+      status: onboardingStatus || ONBOARDING_STATUS_ENUM.CLOSED,
+    });
+  };
+
+  // account sidebar
+  const [isAccountSidebarOpened, setIsAccountSidebarOpened] = useState(false); // or maybe use ux plugin?
+
+  const openAccountSidebar = () => {
+    setIsAccountSidebarOpened(true);
+  };
+
+  const closeAccountSidebar = () => {
+    setIsAccountSidebarOpened(false);
+  };
+
+  // navigation sidebar
+  const [isNavigationSidebarOpened, setIsNavigationSidebarOpened] = useState(
+    false,
+  ); // or maybe use ux plugin?
+
+  const openNavigationSidebar = () => {
+    setIsNavigationSidebarOpened(true);
+  };
+
+  const closeNavigationSidebar = () => {
+    setIsNavigationSidebarOpened(false);
   };
 
   useEffect(() => {
-    reketInstance
-      .get(`/me/preferences/manager/${preferenceKey}`)
-      .then(({ value }) => {
-        setIsBeta(value === 'true');
+    onboardingHelper
+      .init()
+      .then((status) => {
+        setOnboardingOpenedState(
+          onboardingHelper.getOpenedStateFromStatus(status),
+        );
       })
-      .catch((error) => {
-        if (error?.status === 404) {
-          setAskBeta(true);
-        } else {
-          throw error;
-        }
-      })
-      .finally(() => setIsLoading(false));
+      .finally(() => {
+        setIsLoading(false);
+      });
   }, []);
 
   pnrContext = {
-    createBetaChoice,
-    askBeta,
-    isBeta,
-    updateBetaChoice,
+    isLoading,
+    // onboarding
```

Comment:
```suggestion
```

Reply:
this comment is there to separate concern in the returned object

Reply:
🤦‍♀️